src/back-end/lib/db/counter.ts

```diff
@@ -0,0 +1,66 @@
+import { tryDb } from 'back-end/lib/db';
+import { valid } from 'shared/lib/http';
+
+interface ViewCounter {
+  name: string;
+  count: number;
+}
+
+export const incrementCounters = tryDb<[string[]], Record<string, number>>(async (connection, names) => {
+  // Update existing counters
+  const existingCounters = await connection<ViewCounter>('viewCounters')
+    .whereIn('name', names)
+    .increment('count')
+    .update({}, 'name');
```

Comment:
Strange that knex requires you to call increment then update.

Reply:
It doesn't require it, but increment only returns # of incremented rows.  I use update here to get the names of the updated rows so I know which counters needed to be inserted in the next step.